Directory.mk

```diff
@@ -26,6 +26,7 @@ SUBDIRS       := $(dir $(wildcard *$(DELIM)Makefile))
 CONFIGSUBDIRS := $(filter-out $(dir $(wildcard *$(DELIM)Kconfig)),$(SUBDIRS))
 CLEANSUBDIRS  += $(dir $(wildcard *$(DELIM).depend))
 CLEANSUBDIRS  += $(dir $(wildcard *$(DELIM).kconfig))
+CLEANSUBDIRS  += $(dir $(wildcard *$(DELIM)*.tar.*))
```

Comment:
Not sure why is this required.  I see the discussion here https://github.com/apache/incubator-nuttx/pull/6586

`.depend` is created in the argtable3 directory and thus `distclean` should execute there and delete the downloaded files.

Reply:
@Ouss4 
'.depend' is NOT created when building the kernel (CONFIG_BUILD_KERNEL=y) only and will be created when we do 'make export'

Reply:
Ah I see.  Thanks for the explanation.

Reply:
But this fix isn't extensible since not all downloaded package contain *.tar.*

Reply:
>But this fix isn't extensible since not all downloaded package contain .tar.

@xiaoxiang781216 
How about creating the .context file when downloading a file such as .tar.gz or generating a new file such as config.h
Then .context will be checked when executing 'make distclean'

Reply:
Yes, I think it's better to generate .context in kernel mode just like other mode.